src/Avalonia.Controls/HotkeyManager.cs

```diff
@@ -102,8 +105,13 @@ static HotKeyManager()
             HotKeyProperty.Changed.Subscribe(args =>
             {
                 var control = args.Sender as IControl;
-                if (args.OldValue != null|| control == null)
+                if (args.OldValue != null || control == null || !(control is ICommandSource))
+                {
+                    Logging.Logger.TryGet(Logging.LogEventLevel.Warning, Logging.LogArea.Control)?.
```

Comment:
I don't think this logic is correct. The warning should only be logged if `!(control is ICommandSource)`, right?

Reply:
Any object does not support the hotkey. Several times I happened to go crazy to remedy an app malfunction, all because the library/toolkit/framework I was using did not report unsupported behaviors in any way.